### PR TITLE
Check for valid token and refresh it when needed

### PR DESCRIPTION
**Pull request by betatim**
Check if we have a token that is valid and if not refresh it.

Related to #14

### CODE REVIEW

File: grading/__main__.py [hunk 1]
```diff
@@ -41,34 +41,55 @@ def init():
...
-        auth = authorize(user, password, scopes, note, note_url,
-                         two_factor_callback=two_factor)
-
+    # check the token we have is still valid by attempting to login with
+    # the token we have if this fails we need a new one
+    if config.get('github', {}).get('token') is not None:
+        try:
+            # We have to use the GitHub API to find out if our login
+            # credentials actually work, this is why we call `me()`
+            gh = gh3.login(token=config['github']['token'])
```
Comment (lwasser) [inline grading/__main__.py]: This looks good @betatim . stupid question. you import github as `GH` in this module but here create a `gh` object. I suppose those two things are different caps vs lowercase but just wondering if that will cause confusion in the future. 

Thank you for catching how to properly call `authorize()`. i somehow got stuck at the deprecated note and missed that they told us how to call it properly.

**Comment by lwasser** [comment]
this looks really good! i had one small question but i think we should  merge it and i will play with it more later tonight. excited actually to have time to work tomorrow before we meet on wednesday!

**Comment by lwasser** [comment]
i will merge this so i can test locally and see if i still get that student clone error i was getting before

File: grading/__main__.py (hunk 1 above)
Comment (betatim) [reply grading/__main__.py]: re: naming. I think it is already a prime area for mistakes. There is `github3` the library that lets us talk to the GitHub API. There is a `github` in this project which contains custom stuff or higher level things for dealing with GitHub. And finally, the way to use `github3` is to create a GitHub object which I often call `gh`.

I don't know a good way to deal with the name collisions other than establishing a good convention and then sticking with it religiously.

**Pull request merged**